src/app/loadout-drawer/loadout-drawer-reducer.ts

```diff
@@ -194,64 +213,80 @@ export function removeItem(
   });
 }
 
+function setEquipForItemInLoadout(
+  defs: D1ManifestDefinitions | D2ManifestDefinitions,
+  item: DimItem,
+  loadout: Loadout,
+  isEquipped: boolean
+) {
+  // Subclasses and some others are always equipped
+  if (singularBucketHashes.includes(item.bucket.hash)) {
+    return;
+  }
+
+  // TODO: it might be nice if we just assigned a unique ID to every loadout item just for in-memory ops like deleting
+  // We can't just look it up by identity since Immer wraps objects in a proxy and getItemsFromLoadoutItems
+  // changes the socketOverrides, so simply search by unmodified ID and hash.
```

Comment:
Agreed. If we operated on the resolved loadout items this'd also be much easier. Maybe that's something to think about for later.

Reply:
This was a comment I copy pasted I think haha. I might actually do this in my next PR.